### ISSUE TITLE
Module binary: Error extracting .tar.gz

### ISSUE THREAD

**Issue by fleimgruber** [dzop/emacs-zmq#20]
Emacs 26.3 on NixOS.
During startup answering the prompt
```
Check for compatible module binary to download? (y or n) y
```
`*Messages*` then shows
```
Downloading https://github.com/dzop/emacs-zmq/releases/download/v0.10.10/emacs-zmq-x86_64-linux-gnu.tar.gz
Verifying sha256 signature of emacs-zmq-x86_64-linux-gnu.tar.gz
Wrote /home/user/.emacs.d/elpa/26.3/develop/zmq-20190812.1910/emacs-zmq-x86_64-linux-gnu.tar.gz
Parsing tar file...
Warning: premature EOF parsing tar file
Extracting ^_\213^H
```
and `*Warnings*` shows
```
Warning (tar): Extracted ‘^_\213^H’, a link, as a normal file
Warning (initialization): An error occurred while loading ‘/home/user/.emacs.d/init.el’:

Args out of range: #<killed buffer>, 513, 27446879344
```
I looked for `/home/user/.emacs.d/elpa/26.3/develop/zmq-20190812.1910/emacs-zmq-x86_64-linux-gnu.tar.gz`, but it is not there (anymore?).

**Comment by fleimgruber** [comment]
I reproducded this just now on a machine running GNU Debian and Emacs 27.1

`*Messages*`

```
Check for compatible module binary to download? (y or n) y
Downloading https://github.com/nnicandro/emacs-zmq/releases/download/v0.10.10/emacs-zmq-x86_64-linux-gnu.tar.gz
Verifying sha256 signature of emacs-zmq-x86_64-linux-gnu.tar.gz
Wrote /home/user/.emacs.d/elpa/27.1/develop/zmq-20200912.1126/emacs-zmq-x86_64-linux-gnu.tar.gz
Parsing tar file... 
Warning: premature EOF parsing tar file
```

and `*Warnings*` shows

```
Warning (tar): Extracted '/home/user/.emacs.d/elpa/27.1/develop/zmq-20200912.1126/^_\213^H', a link, as a normal file
Warning (initialization): An error occurred while loading '/home/user/.emacs.d/init.el':

Args out of range: #<killed buffer>, 513, 27446879344
```

**Comment by nnicandro** [comment]
This is probably fixed by 790033363cf0e78c616cfe117a2f681381e96f29 assuming one of `curl` or `wget` is used to download the module. I'll reopen if you find that there are still issues.
